MPM/mpm_settings.php

```diff
@@ -333,6 +333,67 @@ public function gateways_add_static($gateways)
 		$head = array_slice($gateways, 0, $pos);
 		$tail = array_slice($gateways, $pos);
 		$gateways = array_merge($head, $mollie_gateways, $tail);
+		return $this->reorderGateways($gateways);
+	}
+
+	/**
+	 * Reorders the method listing on the front-end.
+	 * For special countries, such as Belgium should order MisterCash as first method
+	 * @param array $gateways
+	 * @return array $gateways
+	 */
+	public function reorderGateways($gateways)
```

Comment:
Lennard,

 

Oh sorry man, die reOrder functie was voor een specifieke klant. Deze moest helemaal niet gecommit worden. 

Sorry dat je je tijd erin hebt gestoken om dat te reviewen. Dat was niet de bedoeling. 

 

Uhm hoe kan ik dat het beste oplossen nu?

 

George.

 

From: Lennard van Gunst [mailto:notifications@github.com] 
Sent: donderdag 30 juli 2015 10:26
To: mollie/WooCommerce <WooCommerce@noreply.github.com>
Cc: George van Engers <vanengers@gmail.com>
Subject: Re: [WooCommerce] Fix compatibility with 2.1 mollie return page (#35)

 

In MPM/mpm_settings.php <https://github.com/mollie/WooCommerce/pull/35#discussion_r35848443> :

> @@ -333,6 +333,67 @@ public function gateways_add_static($gateways)
>              $head = array_slice($gateways, 0, $pos);
>              $tail = array_slice($gateways, $pos);
>              $gateways = array_merge($head, $mollie_gateways, $tail);
> +            return $this->reorderGateways($gateways);
> +     }
> +
> +     /**
> +      * Reorders the method listing on the front-end.
> +      * For special countries, such as Belgium should order MisterCash as first method
> +      * @param array $gateways
> +      * @return array $gateways
> +      */
> +     public function reorderGateways($gateways)

Mollie sorteert de betaalmethodes zelf ook al aan de hand van de locale. https://github.com/vanengers/WooCommerce/blob/master/MPM%2Fmpm_settings.php#L523 stuurt sinds kort ook de locale mee. Bij 'be', 'nl_BE' of 'fr_BE' is de volgorde bijvoorbeeld:
creditcard, mistercash, belfius, paypal, sofort, ideal, banktransfer, bitcoin, paysafecard

Ik vraag mij alleen af of er in WordPress een specifieke Belgische locale is.

Ik zie in de methode geen volgorde per locale / land. Wordt de volgorde bijvoorbeeld voor Nederlandse klanten nu ook veranderd? Verder is deze code lastig te begrijpen. Is deze methode echt nodig? Is dit voor een specifieke Belgische klant?

In versie 2.x zijn alle gateways zelf te orderen via de standaard WooCommerce functionaliteit. Dit zal dus een tijdelijke fix zijn voor 1.x

—
Reply to this email directly or view it on GitHub <https://github.com/mollie/WooCommerce/pull/35/files#r35848443> .  <https://github.com/notifications/beacon/ADrkE52NUMMNeKGhgqFyN-zfRygU8UX1ks5oidcIgaJpZM4FiEpl.gif> 

  _____  

Geen virus gevonden in dit bericht.
Gecontroleerd door AVG - www.avg.com <http://www.avg.com> 
Versie: 2015.0.6086 / Virusdatabase: 4392/10331 - datum van uitgifte: 07/29/15